main.tf

```diff
@@ -1,15 +1,19 @@
 locals {
   enabled = module.this.enabled
 
-  partition = join("", data.aws_partition.current[*].partition)
+  partition = one(data.aws_partition.current[*].partition)
+
+  deployed_cluster_identifier = local.enabled ? coalesce(one(aws_rds_cluster.primary[*].id), one(aws_rds_cluster.secondary[*].id)) : ""
+  db_subnet_group_name        = one(aws_db_subnet_group.default[*].name)
+  instance_class              = var.serverlessv2_scaling_configuration != null ? "db.serverless" : var.instance_type
 
   cluster_instance_count   = local.enabled ? var.cluster_size : 0
   is_regional_cluster      = var.cluster_type == "regional"
   is_serverless            = var.engine_mode == "serverless"
   is_serverless_v2         = var.instance_type == "db.serverless" && (contains(["aurora-postgresql", "aurora-mysql"], var.engine)) && var.engine_mode == "provisioned"
   enable_http_endpoint     = var.enable_http_endpoint && (local.is_serverless || local.is_serverless_v2)
   ignore_admin_credentials = var.replication_source_identifier != "" || var.snapshot_identifier != null
-  reserved_instance_engine = split("-", var.engine)[1]
+  reserved_instance_engine = var.engine
```

Comment:
I haven't used this but will soon through this module. Wondering if this is intended / or what is the rationale for this?

In the examples complete, I'm seeing engine is passed in as `aurora-postgres` or `aurora-mysql`
and in the Terraform docs
![image](https://github.com/user-attachments/assets/bbbb1ccd-7611-4f01-ac35-ae3006578311)

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_reserved_instance

and https://github.com/cloudposse/terraform-aws-rds-cluster/blob/04c61c2c00cf19c0708e98b5b574ff50b4460a78/main.tf#L105

Reply:
The product description is supposed to match the engine exactly. `aurora-mysql` and `mysql` are separate available engines.

Reply:
Ahhh I totally forgot about that. Thanks for pointing that out!